code-challenges.test.js

```diff
@@ -10,47 +12,125 @@
-const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"]
-const letterE = "e"
+const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"];
+const letterE = "e";
 // Expected output: ["Cherry", "Blueberry", "Peach"]
 
+describe("wordReturner", () => {
+  it("takes in an array of words and a single letter and returns all the words that contain that particular letter", () => {
+    const arrayOfWords1 = ["Apple", "Banana", "Plum", "Orange", "Kiwi"];
+    const letterA = "a";
+    const arrayOfWords2 = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach"];
+    const letterE = "e";
+
+    expect(wordReturner(arrayOfWords1)).toEqual(["Apple", "Banana", "Orange"]);
+    expect(wordReturner(arrayOfWords2)).toEqual([
+      "Cherry",
+      "Blueberry",
+      "Peach",
+    ]);
+  });
+});
+
+// ReferenceError: wordReturner is not defined
 
 // b) Create the function that makes the test pass.
 
+// PSEUDOCODE:
+// create function called wordReturner
+// takes in array and single letter
+// create for loop to iterate
+// if statement
+// check each index for the letter provided
+// true, return the word
+// false, don't
+// return array
+
+// function wordReturner(array,value) {
+//     for (let i = 0; i < array.length; i++) {
+//         if (array[i].indexOf(value && value.toUpperCase())) {
+//             return array[i]
+//          }
+//     }
+// }
+
+// // Expected: ["Apple", "Banana", "Orange"]
+// // Received: "Apple"
+// // Not returning the rest of the words for some reason..
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is exactly one pair and one three of a kind.
 
 // a) Create a test with expect statements using the variable provided.
 
-const hand1 = [5, 5, 5, 3, 3]
+const hand1 = [5, 5, 5, 3, 3];
 // Expected output: true
-const hand2 = [5, 5, 3, 3, 4]
+const hand2 = [5, 5, 3, 3, 4];
 // Expected output: false
-const hand3 = [5, 5, 5, 5, 4]
+const hand3 = [5, 5, 5, 5, 4];
 // Expected output: false
 
+describe("fullHouse", () => {
+  it("takes in an array of 5 numbers and determines whether or not the array is a 'full house'", () => {
+    const hand1 = [5, 5, 5, 3, 3];
+    const hand2 = [5, 5, 3, 3, 4];
+    const hand3 = [5, 5, 5, 5, 4];
+    expect(fullHouse(hand1)).toEqual([5, 5, 5, 3, 3]);
+    expect(fullHouse(hand2)).toEqual([5, 5, 3, 3, 4]);
+    expect(fullHouse(hand3)).toEqual([5, 5, 5, 5, 4]);
+  });
+});
+
+// ReferenceError: fullHouse is not defined, good fail
 
 // b) Create the function that makes the test pass.
+
+// PSEUDOCODE:
+// create fn named fullHouse
+    // takes in array
+// check array for 3 of same number && 2 of different number
+    // iterate using for loop
+    // slice and return true values 
+
+function fullHouse(array) {
+    let num1 = num
+    for (let i = 0; i < array.length; i++) {
+        if (array[i] == array.every(num1)) {
+            return array[i]
```

Comment:
Again the "return" inside of this for loop means you are canceling anything past the first iteration of this for loop.